### ISSUE TITLE
Bug: Toggle-behavior with NodeJS, restart flag and custom signal

### ISSUE THREAD

**Issue by c7hm4r** [watchexec/watchexec#426]
I have a text file `a` and following NodeJS script `watchexec.js`:

```js
const process = require("node:process");
/* const readline = require("node:readline"); */

let a = 0;
const i = setInterval(() => console.log(a++), 1000);

/*
if (process.platform === "win32") {
  const readLineInterface = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });
  readLineInterface.on("SIGINT", () => {
    process.emit("SIGINT", "SIGINT");
  });
}
*/
process.on("SIGINT", () => {
  console.log("SIGINT");
  clearInterval(i);
});
```

Basically it implements a graceful exit on SIGINT.

### Repro steps / actual behavior

1. Run Command: `watchexec -n -r -s SIGINT -w a -- node watchexec.js`. HTOP shows a NodeJS child process of watchexec.

2. Then when I run `touch a`, the script logs `SIGINT` but is not restarted. HTOP does not show a NodeJS child process of watchexec anymore.

3. When I run `touch a` again, the script is started again. HTOP shows a NodeJS child process of watchexec again.

4. Goto 2

### Expected behavior

It should restart the script also after every first/third/fifth/… `touch a` using SIGINT.

### Versions

NodeJS: v18.9.0
OS: Fedora 36 x64
watchexec: 1.20.6

### Output with -vvv

There seems to be another race condition, because (irregularly) only sometimes it shows an error, but it has seemingly no effect on the described start/stop behavior.

[watchexec_output.txt](https://github.com/watchexec/watchexec/files/9735358/watchexec_output.txt)

**Comment by passcod** [comment]
Hmm, yes, I think there's a bug where in signal+restart mode, it will wait 60s after sending the signal before restarting. Does the script start after 60s when doing `touch a`?

**Comment by c7hm4r** [comment]
> Does the script start after 60s when doing `touch a`?

Yes, exactly.

```
2022-10-08T21:05:25.325526Z ERROR watchexec_cli::config::init: io(waiting on process group): No child processes (os error 10)
2022-10-08T21:06:25.322189Z DEBUG watchexec::action::outcome_worker: outcome worker finished outcome=Both(Both(Signal(Interrupt), Both(Sleep(60s), Stop)), Start) gen=2
2022-10-08T21:06:25.322251Z DEBUG supervisor: watchexec::command::supervisor: new
2022-10-08T21:06:25.322271Z DEBUG supervisor: watchexec::command::supervisor: preparing command grouped=true command=Exec { prog: "node", args: ["watchexec.js"] }
2022-10-08T21:06:25.322299Z DEBUG supervisor: watchexec::command::supervisor: running pre-spawn handler
2022-10-08T21:06:25.322448Z DEBUG watchexec_cli::config::runtime: inserting environment variable k="WATCHEXEC_META_CHANGED_PATH" v="a"
2022-10-08T21:06:25.322478Z DEBUG watchexec_cli::config::runtime: inserting environment variable k="WATCHEXEC_WRITTEN_PATH" v="a"
2022-10-08T21:06:25.322518Z DEBUG watchexec_cli::config::runtime: inserting environment variable k="WATCHEXEC_COMMON_PATH" v="/home/christoph/data/temp"
2022-10-08T21:06:25.322561Z DEBUG supervisor: watchexec::command::supervisor: spawning command command=Command { std: "node" "watchexec.js", kill_on_drop: true }
2022-10-08T21:06:25.324663Z DEBUG supervisor: watchexec::command::supervisor: process group spawned pgid=12732
2022-10-08T21:06:25.324713Z DEBUG supervisor: watchexec::command::supervisor: running post-spawn handler
2022-10-08T21:06:25.324789Z DEBUG supervisor: watchexec::command::supervisor: spawned process process=Grouped(AsyncGroupChild) pid=12732
0
1
2
```

**Comment by passcod** [comment]
Right, so, I know what's happening, but to fix it I need a new outcome primitive, roughly "do A until/unless B happens", and I'm not quite sure how to do that yet. I'll think about it.

**Comment by passcod** [comment]
I could probably add an option to disable or reduce the 60s wait (which is there to do a SIGKILL if the signal failed to stop the program), as a workaround until the bug is actually fixed.

**Comment by c7hm4r** [comment]
Thanks for working on that. Though, it's not that urgent for me, so don't waste time/energies.

**Comment by passcod** [comment]
This was since fixed, sometime about 1.22.1.
